cyberdrop_dl/crawlers/pkmncards.py

```diff
@@ -0,0 +1,244 @@
+        scrape_item.setup_as_profile("")
+        page_url = self.primary_base_domain / "series" / scrape_item.url.parts[2]
+        await self._iter_from_url(scrape_item, page_url)
+
+    @error_handling_wrapper
+    async def card_set(self, scrape_item: ScrapeItem) -> None:
+        scrape_item.setup_as_album("")
+        page_url = self.primary_base_domain / "set" / scrape_item.url.parts[2]
+        await self._iter_from_url(scrape_item, page_url)
+
+    async def _iter_from_url(self, scrape_item: ScrapeItem, url: URL) -> None:
+        page_url = url.with_query(sort="date", ord="auto", display="images")
+        async for soup in self.web_pager(page_url):
+            for thumb in soup.select(CARD_SELECTOR):
+                parts: tuple[str, str, str] = thumb.select_one("img")["src"], thumb["href"], thumb["title"]  # type: ignore
+                link_str, card_page_url_str, title = parts
+                card_page_url = self.parse_url(card_page_url_str)
+                download_url = self.parse_url(link_str)
+                simple_card = create_simple_card(title, download_url)
+                new_scrape_item = scrape_item.create_child(card_page_url)
+                self.manager.task_group.create_task(self.handle_simple_card(new_scrape_item, simple_card))
+                scrape_item.add_children()
+
+    @error_handling_wrapper
+    async def card(self, scrape_item: ScrapeItem) -> None:
+        async with self.request_limiter:
+            soup: BeautifulSoup = await self.client.get_soup(self.domain, scrape_item.url)
+
+        name = soup.select_one(CARD_NAME_SELECTOR).text  # type: ignore
+        number = soup.select_one(CARD_NUMBER_SELECTOR).text  # type: ignore
+        link_str: str = soup.select_one(CARD_DOWNLOAD_SELECTOR)["href"]  # type: ignore
+        link = self.parse_url(link_str)
+        card_set = create_set(soup)
+        card = Card(name, number, card_set, link)
+        await self.handle_card(scrape_item, card)
+
+    @error_handling_wrapper
+    async def handle_card(self, scrape_item: ScrapeItem, card: Card) -> None:
+        if not card.name:
+            raise ScrapeError(422)
+        link = card.download_url  # .with_suffix(".png")  # they offer both jpg and png. png is higher quality
+        set_title = self.create_title(f"{card.set.name} ({card.set.full_code})")
+        scrape_item.setup_as_album(set_title, album_id=card.set.abbr)
+        scrape_item.possible_datetime = card.set.release_date
+        filename, ext = self.get_filename_and_ext(link.name, assume_ext=".jpg")
+        custom_filename, _ = self.get_filename_and_ext(f"{card.full_name}{ext}")
+        await self.handle_file(link, scrape_item, filename, ext, custom_filename=custom_filename)
+
+    async def handle_simple_card(self, scrape_item: ScrapeItem, simple_card: SimpleCard):
+        @create_task_id
+        @error_handling_wrapper
+        async def get_card_set(self, scrape_item: ScrapeItem):
```

Comment:
```suggestion
        async def get_card_set(scrape_item: ScrapeItem):
```
`get_card_set` doesn't need `self` as it has the context from `handle_simple_card`.

Reply:
It's required for the `@error_handling_wrapper` and `@create_task_id` to work. The first argument must be self.